rust/connlib/snownet/src/node.rs

```diff
@@ -1975,6 +1946,47 @@ where
         }
     }
 
+    fn handle_tunnel_timeout(
+        &mut self,
+        now: Instant,
+        allocations: &mut BTreeMap<RId, Allocation>,
+        transmits: &mut VecDeque<Transmit<'static>>,
+    ) {
+        // Don't update wireguard timers until we are connected.
+        let Some(peer_socket) = self.socket() else {
+            return;
+        };
+
+        /// [`boringtun`] requires us to pass buffers in where it can construct its packets.
+        ///
+        /// When updating the timers, the largest packet that we may have to send is `148` bytes as per `HANDSHAKE_INIT_SZ` constant in [`boringtun`].
+        const MAX_SCRATCH_SPACE: usize = 148;
+
+        let mut buf = [0u8; MAX_SCRATCH_SPACE];
+
+        match self.tunnel.update_timers_at(&mut buf, now) {
+            TunnResult::Done => {}
+            TunnResult::Err(WireGuardError::ConnectionExpired) => {
+                tracing::info!("Connection failed (wireguard tunnel expired)");
+                self.state = ConnectionState::Failed;
+            }
+            TunnResult::Err(e) => {
+                tracing::warn!(?e);
```

Comment:
Are there other errors that can pop up that we should be moving to a ConnectionFailed state for?

Reply:
Not that I am aware of. Note that this code is old and only go moved so this has been working fine for months!